polars/query.py

```diff
@@ -352,18 +350,14 @@
             .group_by("k")
             .agg(
                 [
-                    pl.col("Referer").map_elements(
-                        lambda y: len(y), return_dtype=pl.Int64
```

Comment:
This was wrong. Didn't compute the `AVG` and it uses a python function, where we have native expressions for this.